services/agora/src/composables/useConversationData.ts

```diff
@@ -38,34 +45,78 @@ export function useConversationData() {
       return null;
     }
 
-    return Array.isArray(route.params.postSlugId)
-      ? route.params.postSlugId[0]
-      : route.params.postSlugId;
+    const postSlugId = (route.params as { postSlugId?: string | string[] })
+      .postSlugId;
+    return Array.isArray(postSlugId) ? postSlugId[0] : postSlugId || null;
   }
 
-  async function loadConversationData(refresh = false) {
+  async function loadConversationData() {
     const slugId = getConversationSlugId();
+    if (!slugId) return false;
 
-    if (slugId) {
-      return await conversationStore.loadConversationData(slugId, refresh);
+    // Wait for auth to be initialized
+    if (!isAuthInitialized.value) {
+      await new Promise<void>((resolve) => {
+        const unwatch = watch(isAuthInitialized, (isReady) => {
+          if (isReady) {
+            unwatch();
+            resolve();
+          }
+        });
+      });
```

Comment:
this feels complicated i wonder if there is a way to rearchitect such that we don't need to do this in individual component

anyhow, it's unrelated to the PR specifically